src/ct_selftest/ct_selftest.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+ * Basic tests for the CT::poison annotations.
+ * Some of those are expected to fail, because they deliberately
+ * branch on secret memory.
+ *
+ * (C) 2024 Jack Lloyd
+ * (C) 2024 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include <iostream>
+
+#include <botan/internal/ct_utils.h>
+
+#if defined(BOTAN_CT_POISON_ENABLED)
```

Comment:
I’d rather we compile everything regardless, with a small ifdef in main checking the macro